examples/preprocessing/plot_target_regressor_encoder.py

```diff
@@ -0,0 +1,121 @@
+# from 18 to 14810:
+numerical_features = ["price"]
+categorical_features = [
+    "country",
+    "province",
+    "region_1",
+    "region_2",
+    "variety",
+    "winery",
+]
+
+X = df[numerical_features + categorical_features]
+y = df["points"]
+X.nunique().sort_values(ascending=False)
+
+# %%
+# We split the dataset into a training and test set:
+from sklearn.model_selection import train_test_split
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)
+
+print(f"Samples in training set: {len(X_train)}\nSamples in test set: {len(X_test)}")
+
+# %%
+# Building and Training Pipelines with Different Encoders
+# =======================================================
+# Dropping the categorical features
+# ---------------------------------
+# As a basline, we construct a pipeline where the categorical features are
+# dropped.
+from sklearn.compose import ColumnTransformer
+from sklearn.pipeline import Pipeline
+from sklearn.ensemble import HistGradientBoostingRegressor
+
+prep = ColumnTransformer(
+    [
+        ("num", "passthrough", numerical_features),
+        ("cat", "drop", categorical_features),
+    ]
+)
+
+reg_drop_cats = Pipeline(
+    [("prep", prep), ("hist", HistGradientBoostingRegressor(random_state=0))]
+)
+reg_drop_cats
+
+# %%
+# Here we train and use the root mean squared error to evalute the baseline
+# model:
+from sklearn.metrics import mean_squared_error
+
+reg_drop_cats.fit(X_train, y_train)
+reg_drop_cats_rmse = mean_squared_error(y_test, reg_drop_cats.predict(X_test))
+print(f"RMSE for dropping categorical features: {reg_drop_cats_rmse:.4}")
+
+# %%
+# Using the OrdinalEncoder
+# ------------------------
+# Since the categorical features have missing values, we impute the feature
+# with `'sk_missing'` before passing it to the :class:`OrdinalEncoder`.
+from sklearn.preprocessing import OrdinalEncoder
+
+cat_prep = OrdinalEncoder(handle_unknown="use_encoded_value", unknown_value=-1)
+
+# %%
+# We modify the original pipeline to use the ordinal categorical preprocessing:
+reg_ordinal = reg_drop_cats.set_params(prep__cat=cat_prep)
+reg_ordinal
+
+# %%
+# When we include the categorical features through ordinal encoding the RMSE
+# improves:
+reg_ordinal.fit(X_train, y_train)
+reg_ordinal_rmse = mean_squared_error(
+    y_test, reg_ordinal.predict(X_test), squared=False
+)
+print(f"RMSE with ordinal encoding: {reg_ordinal_rmse:.4}")
+
+# %%
+# Using the TargetRegressorEncoder
+# --------------------------------
+# Finally, we replace the ordinal encoder with the
+# :class:`TargetRegressorEncoder`:
+from sklearn.preprocessing import TargetRegressorEncoder
+
+reg_target = reg_ordinal.set_params(prep__cat=TargetRegressorEncoder())
+reg_target
+
+# %%
+# The :class:`TargetRegressorEncoder` further improves the RMSE:
+reg_target.fit(X_train, y_train)
+reg_target_rmse = mean_squared_error(y_test, reg_target.predict(X_test), squared=False)
+print(f"RMSE with target encoding: {reg_target_rmse:.4}")
```

Comment:
Can we have some kind of wrap up/conclusion? Something like "As expected the performance improves when using the additional categorical features, compared to using only the price of the wine to predict its rating. The `TargetRegressorEncoder` further improves the performance because it does not assign an order to the categorical features, which in reality do not have an order."

I don't know if this is the reason why the performance improves, so just a guess. Does anyone know "the" reason?